src/components/timers/XY.tsx

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useRef, useState } from "react";
+import { useTimers } from "../../context/TimerContext";
+
+import Button from "../generic/Button";
+import { StyledButtonContainer, TimerDisplay, TimerContainer } from "../generic/ContainerDisplays";
+import { formatTime } from "../../utils/helpers";
+
+const XY = ({ id }: { id: string }) => {
+  const { timers, updateTimerTimeLeft, updateTimerState, nextTimer, removeTimer } = useTimers();
+  const timer = timers.find((t) => t.id === id);
+
+  if (!timer) return null;
+
+  const { minutes = 0, seconds = 0, numberOfRounds } = timer.config;
+  const isRunning = timer.state === "running";
+
+  const initialTime = (minutes * 60 + seconds) * 1000;
+  const [roundsLeft, setRoundsLeft] = useState<number>(numberOfRounds);
+  const timeLeft = timer.timeLeft === undefined ? initialTime : timer.timeLeft; // Default to initialTime if undefined
+
+  const intervalRef = useRef<number | null>(null);
+
+  useEffect(() => {
+
+    if (isRunning && timeLeft > 0) {
+      intervalRef.current = setInterval(() => {
+        updateTimerTimeLeft(id, timeLeft - 10);
+      }, 10);
+    } else if (isRunning && timeLeft <= 0 && roundsLeft > 1) {
+      if (intervalRef.current !== null) {
+        clearInterval(intervalRef.current as number);
+      } setRoundsLeft((prevRounds) => prevRounds - 1);
+      updateTimerTimeLeft(id, initialTime);
+    } else if (isRunning && timeLeft <= 0 && roundsLeft <= 1) {
+      if (intervalRef.current !== null) {
+        clearInterval(intervalRef.current as number);
+      } updateTimerState(id, "completed");
+      nextTimer();
+    }
+
+    return () => {
+      if (intervalRef.current !== null) {
+        clearInterval(intervalRef.current as number);
+      }
+    };
+  }, [isRunning, timeLeft, roundsLeft]);
+
+
+  return (
+    <TimerContainer>
```

Comment:
I like what you're doing with the shared components here, and you can take it one step further in each of these timers to DRY up the code even more. It seems like all four of the timers have a TimerDisplay and remove Button. You could consider creating a generic timer component that has those elements in there, and then you could pass in the parts that are unique to the timers (like below, the `<div>Rounds Remaining: {roundsLeft}</div>` component) either as children or as a prop itself. 

ex. 
```
<GenericTimer timeLeft={timeLeft} onRemove={() => removeTimer(id)}>
<div>Rounds Remaining: {roundsLeft}</div>
</GenericTimer>

// which could look like 
const GenericTimer = ({timeLeft, onRemove, children) => {
...
}
```